src/__tests__/poller.test.ts

```diff
@@ -0,0 +1,141 @@
+import Poller from '../poller'
+import type { Options } from '../types'
+import { getRandom } from '../utils'
+
+jest.useFakeTimers()
+
+jest.mock('../utils.ts', () => ({
+  getRandom: jest.fn(),
+  logError: jest.fn()
+}))
+
+interface PollerArgs {
+  fetchFlags: jest.MockedFunction<() => Promise<any>>
+  configurations: Partial<Options>
+}
+
+interface TestArgs {
+  delayFunction: jest.MockedFunction<() => number>
+  logSpy: jest.SpyInstance
```

Comment:
I've left the Spy and its mockImplementation in the suite, so that log statements aren't printed to console, but no more assertions on it.